receiver/statsdreceiver/config.go

```diff
@@ -63,10 +64,22 @@ func (c *Config) validate() error {
 		}
 
 		switch eachMap.ObserverType {
-		case protocol.GaugeObserver, protocol.SummaryObserver:
+		case protocol.GaugeObserver, protocol.SummaryObserver, protocol.HistogramObserver:
 		default:
 			errs = multierr.Append(errs, fmt.Errorf("observer_type is not supported: %s", eachMap.ObserverType))
 		}
+
+		if eachMap.ObserverType == protocol.HistogramObserver {
+			if eachMap.Histogram.MaxSize != 0 && (eachMap.Histogram.MaxSize < structure.MinSize || eachMap.Histogram.MaxSize > structure.MaximumMaxSize) {
```

Comment:
Are we (reasonably) sure that the OTel Go implementation range of accepted sizes will cover this range too? Should we be more restrictive to avoid breaking changes in the future if we go with a more restricted range on OTel Go?

Reply:
OTel-Go will not be accepting these, it's the consumers (e.g., other OTel collector processors & exporters and their downstream receivers). There is not a specified limit to this number and we could remove this check. This limit is 100x the SDK-specified default size--if the OTLP protocol were changed to specify a limit, I suppose we'd update this code.

Reply:
Not sure I follow here, do you mean that the OpenTelemetry Go SDK will not impose a maximum/minimum size for ExponentialHistograms then?

Reply:
As for minimum size, there's no point in a histogram with less than 2 buckets, so I set that as the minimum max-size. 

There isn't an OTel-specified maximum in the protocol or the SDK specification. I'm not opposed to such a limit being specified, I'm just not sure it's very important to do. An analogous situation in the trace SDK would be that we allow a limit on attributes count to be set but we don't limit the maximum number of attributes before dropping afaik. As a vendor, we have a realistic limit set for span attribute count and will reject spans with too many attributes, but trying to get everyone to agree to an upper bound seems difficult. The integer-width-imposed limit of 2**31 is much too high.

For the expo-histo specifically, we didn't place a limit on "scale" either. It's possible to have meaningful data at extreme scales with the existing protocol and someone might find a use for that. The underlying data structure sets a limit on scale at 20 simply because it's difficult to test beyond 20 (b/c even `math/big` has limits). Presumably we could extend the protocol to support 128-bit precision (i.e., greater max scale) or more than 2**31 entries (i.e,. greater max size) in the future.

Reply:
Thanks for the additional details. I think it's safe to assume then that we can keep support for the same range when we switch to use the OTel Go SDK :+1: